remi/server.py

```diff
@@ -172,7 +173,10 @@ def bytetonum(self,b):
 
     def read_next_message(self):
         debug_message('read_next_message\n')
-        length = self.rfile.read(2)
+        try:
+            length = self.rfile.read(2)
+        except timeout:
```

Comment:
Yes I am testing it now ;-)

wrt. importing, I never really optimize for start-up (i.e. import time)
performance, and instead optimize for readability. I will go through and
clean all imports for remi eventually, but maybe after I have conviced
everyone to use six to manage py2/3 compatibility ;-)

On 12 November 2015 at 12:36, Sammy Pfeiffer <notifications@github.com>
wrote:

> In remi/server.py
> <https://github.com/dddomodossola/remi/pull/66#discussion_r44647412>:
>
> > @@ -172,7 +173,10 @@ def bytetonum(self,b):
> >
> >      def read_next_message(self):
> >          debug_message('read_next_message\n')
> > -        length = self.rfile.read(2)
> > +        try:
> > +            length = self.rfile.read(2)
> > +        except timeout:
>
> I considered it, but then I need to do:
>
> import socket
>
> Up on the file, which may import many things. I wanted to skip that. If
> you consider that is better to do it that way I'll do.
>
> But please, before even considering it, try it with some code you have for
> a bit of time! This is quite a critical point in the code and there may be
> a reason this was this way (other than technological debt).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dddomodossola/remi/pull/66/files#r44647412>.
>